src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -10,40 +10,75 @@
 import edu.wpi.first.wpilibj2.command.StartEndCommand;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.Constants;
+import frc.robot.stateManagement.PlacementMode;
 
 public class IntakeSubsystem extends SubsystemBase {
-
   private CANSparkMax intakeMotor;
+  private CANSparkMax chooserMotor;
 
   public IntakeSubsystem() {
     intakeMotor = new CANSparkMax(Constants.IntakeConstants.INTAKE_MOTOR_ID, MotorType.kBrushless);
+    chooserMotor =
+        new CANSparkMax(Constants.IntakeConstants.INTAKE_CHOOSER_ID, MotorType.kBrushless);
     intakeMotor.restoreFactoryDefaults();
+    chooserMotor.restoreFactoryDefaults();
     intakeMotor.setSmartCurrentLimit(40);
+    chooserMotor.setSmartCurrentLimit(20);
+  }
+
+  // TODO: Add check to make sure turret is below 45 degrees before running & add photogate when
+  // implemented.
+  public void runIntake() {
+    intakeMotor.set(Constants.IntakeConstants.INTAKE_PERCENTAGE);
+  }
+
+  public void reverseIntake() {
+    intakeMotor.set(-Constants.IntakeConstants.INTAKE_PERCENTAGE);
+  }
+
+  public void speakerChooser() {
+    chooserMotor.set(Constants.IntakeConstants.CHOOSER_PERCENTAGE);
+  }
+
+  public void ampChooser() {
+    chooserMotor.set(-Constants.IntakeConstants.CHOOSER_PERCENTAGE);
+  }
+
+  public void speakerIntake() {
+    runIntake();
+    speakerChooser();
+  }
+
+  public void ampIntake() {
+    runIntake();
+    ampChooser();
+  }
+
+  public void stopMotors() {
+    chooserMotor.set(0);
+    intakeMotor.set(0);
   }
 
-  public Command intakeCommand() {
-    // Inline construction of command goes here.
-    // Subsystem::RunOnce implicitly requires `this` subsystem.
-    return runOnce(
-        () -> {
-          /* one-time action goes here */
-        });
+  public Command reverseIntakeCommand() {
+    return new StartEndCommand(this::reverseIntake, this::stopMotors, this);
```

Comment:
Does the chooser need to reverse as well ? i.e. run the opposite of the current placement

Reply:
Coast mode should be fine.